ddtrace/internal/rate_limiter.py

```diff
@@ -38,54 +38,57 @@ def __init__(self, rate_limit):
         self.tokens = rate_limit  # type: float
         self.max_tokens = rate_limit
 
-        self.last_update = compat.monotonic()
+        self.last_update_ns = compat.monotonic_ns()
 
-        self.current_window = 0  # type: float
+        self.current_window_ns = 0  # type: float
         self.tokens_allowed = 0
         self.tokens_total = 0
         self.prev_window_rate = None  # type: Optional[float]
 
         self._lock = threading.Lock()
 
-    def is_allowed(self):
-        # type: () -> bool
+    def is_allowed(self, timestamp_ns=None):
```

Comment:
I don't see a reason why this should be optional

Reply:
the only reason I did that was for backwards compatibility, but since this is only used by the sampler and internal, should be safe to change.